ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -12068,8 +12068,9 @@ void HqlCppTranslator::buildScriptFunctionDefinition(BuildCtx &ctx, IHqlExpressi
     else
         embedOptions.setown(getEmbedOptionString(bodyCode));
     IValue *optionVal = embedOptions->queryValue();
+    bool isActivity = functionBodyIsActivity(bodyCode);
 
-    bool threadlocal = bodyCode->hasAttribute(_threadlocal_Atom) && queryVal != nullptr && optionVal != nullptr;
+    bool threadlocal = bodyCode->hasAttribute(_threadlocal_Atom) && queryVal != nullptr && optionVal != nullptr && !isActivity;
```

Comment:
This isn't necessarily the perfect fix - we could pass in the activity separately instead, and then still keep the threadlocal. But it's probably good enough and keeps things compatible. When using activity flag the chances are it's not being reused a lot anyway.